keras/engine/base_layer.py

```diff
@@ -2199,21 +2199,24 @@ def _infer_output_signature(self, inputs, args, kwargs, input_masks):
       input_masks = tf.nest.map_structure(
           keras_tensor.keras_tensor_to_placeholder, input_masks)
 
-      with backend.name_scope(self._name_scope()):  # pylint: disable=not-callable
-        with autocast_variable.enable_auto_cast_variables(
-            self._compute_dtype_object):
-          # Build layer if applicable (if the `build` method has been
-          # overridden).
-          # TODO(kaftan): do we maybe_build here, or have we already done it?
-          self._maybe_build(inputs)
-          inputs = self._maybe_cast_inputs(inputs)
-          outputs = call_fn(inputs, *args, **kwargs)
+      ctx = context.context()
+      assert ctx is not None
+      with ops.name_scope(ctx.scope_name[:-1]):
+        with backend.name_scope(self._name_scope()):  # pylint: disable=not-callable
```

Comment:
If wonder if stacking these 2 name scopes may have unexpected side effects. Why not make the outer scope part of `self._name_scope`?